### PR TITLE
Add documentation for various internal/external functions

### PR DESCRIPTION
**Pull request by asn-d6**

### CODE REVIEW

**Comment by xrchz** [comment]
Amazing - thanks!

Failing tests need to be fixed - I think one of the renames wasn't complete.

File: src/c_kzg_4844.c [hunk 1]
```diff
@@ -1109,6 +1263,17 @@ C_KZG_RET compute_aggregate_kzg_proof(KZGProof *out,
     return ret;
 }
 
+/**
+ * Computes the aggregate KZG proof for multiple blobs.
+ *
+ * @param[out] out Pointer to the KZGProof object to store the result in.
```
Comment (xrchz) [inline src/c_kzg_4844.c:1269]: it's just a boolean

Comment (asn-d6) [reply src/c_kzg_4844.c]: ugh. fixed in bd67d53

**Comment by jtraglia** [comment]
One nit, there aren't tabs between tag, name, description.
```
@param[in] x a value
```
vs
```
@param[in] x    a value
```

**Comment by xrchz** [comment]
I hope you mean alignment with spaces and not literal tabs?

**Comment by jtraglia** [comment]
Ah yes, that's what I mean.

**Comment by asn-d6** [comment]
My editor does not do this in-comment alignment and it's a tedious job to do this. Is there a linter one can use, or can your editor do it automatically?

**Comment by xrchz** [comment]
I can do the alignment thing after we merge. Can you fix the conflicts though (probably just rebase over latest commit)?

File: src/c_kzg_4844.c [hunk 2]
```diff
@@ -767,7 +842,15 @@ static void poly_lincomb(Polynomial *out, const Polynomial *vectors, const fr_t
     }
 }
 
-
+/**
+ * Compute and return [ x^0, x^1, ..., x^{n-1} ]
+ *
+ * @remark `out` is left untouched if n == 0
```
Comment (jtraglia) [inline src/c_kzg_4844.c:848]: Missing punctuation. First sentence & the remark.

File: src/c_kzg_4844.c [hunk 3]
```diff
@@ -848,6 +960,18 @@ C_KZG_RET blob_to_kzg_commitment(KZGCommitment *out, const Blob *blob, const KZG
 static C_KZG_RET verify_kzg_proof_impl(bool *out, const g1_t *commitment, const fr_t *x, const fr_t *y,
                                        const g1_t *proof, const KZGSettings *ks);
 
+/**
+ * Verify a KZG proof claiming that p(z) == y.
```
Comment (jtraglia) [inline src/c_kzg_4844.c:964]: Do you want to wrap `p(z)==y` in backticks here?

File: src/c_kzg_4844.c [hunk 4]
```diff
@@ -989,6 +1130,19 @@ static C_KZG_RET compute_aggregated_poly_and_commitment(Polynomial *poly_out, g1
     return C_KZG_OK;
 }
 
+/**
+ * Computes aggregate KZG proof given for multiple blobs
```
Comment (jtraglia) [inline src/c_kzg_4844.c:1134]: Missing punctuation.

File: src/c_kzg_4844.c [hunk 5]
```diff
@@ -871,23 +995,23 @@ C_KZG_RET verify_kzg_proof(bool *out,
 }
 
 /**
- * Check a KZG proof at a point against a commitment.
+ * Helper function: Verify KZG proof claiming that p(z) == y
```
Comment (jtraglia) [inline src/c_kzg_4844.c:998]: Missing punctuation and maybe wrap `p(z)==y` with backticks.

**Review by jtraglia — CHANGES_REQUESTED** [review]
Overall looks good. Thanks for this! Just a few nits.

File: src/c_kzg_4844.c (hunk 2 above)
Comment (jtraglia) [reply src/c_kzg_4844.c:848]: Oh also, for consistency I think you should wrap `n==0` in backticks.

**Review by jtraglia — APPROVED** [review]
LGTM thanks!

**Pull request merged**